src/BlockTypes/NoResults.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use Automattic\WooCommerce\Blocks\Utils\ProductCollectionUtils;
+
+/**
+ * NoResults class.
+ */
+class NoResults extends AbstractBlock {
+
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'no-results';
```

Comment:
May be worth adding a prefix suggesting this block belongs to Product Collection. What do you think?

```suggestion
	protected $block_name = 'product-collection-no-results';
```

Reply:
Agreed! Made the changes in commit https://github.com/woocommerce/woocommerce-blocks/pull/11783/commits/9dc01a9b4adce5cde28645de37bc4d791bd8cbcd.